### PR TITLE
Several small fixes to the docs

### PR DESCRIPTION
**Pull request by xylar**
This merge also adds a table of contents to design-doc template, following the example from time stepping.

Checklist
* [x] Documentation:
  * [x] Design document has been updated
  * [x] User's Guide has been updated
  * [x] Developer's Guide has been updated
  * [x] Documentation has been [built locally](https://e3sm-project.github.io/Omega/develop/devGuide/BuildDocs.html) and changes look as expected

### CODE REVIEW

File: components/omega/doc/design/TimeStepping.md [hunk 1]
```diff
@@ -115,4 +115,4 @@ $$
 
 Where the initial conditions for $\boldsymbol{u}$ and $\phi$ are uniformly one, the tracer restoring value in this case is $\phi_0=0$, and the solution is simply exponential decay to zero.
 
-The timestepping is an intrinsic part of the Omega-0 model, so the verification and testing are identical to those in the [Omega V0: Shallow Water](OmegaV0ShallowWater.md) design document. The timestepping will be tested with convergence tests in time using the manufactured solution test case.
+The timestepping is an intrinsic part of the Omega-0 model, so the verification and testing are identical to those in the {ref}`omega-design-shallow-water-omega0` design document. The timestepping will be tested with convergence tests in time using the manufactured solution test case.
```
Comment (philipwjones) [inline components/omega/doc/design/TimeStepping.md:118]: Out of curiousity - is this now the preferred form for references within a doc (as opposed to the [text](ref) form) ?

**Review by philipwjones — APPROVED** [review]
Built and viewed the docs - all looks good. Thanks for the fixes.

File: components/omega/doc/design/TimeStepping.md (hunk 1 above)
Comment (xylar) [reply components/omega/doc/design/TimeStepping.md:118]: Yes, this is my preference (this way, references are the same whether it's to the top of the document or somewhere internally) but the other way is fine, too.  In this case, the file name was wrong so I decided to switch to my preferred approach.

**Comment by xylar** [comment]
@philipwjones, feel free to merge this when convenient.

**Pull request merged**